openedx_external_enrollments/external_enrollments/greenfig_external_enrollment.py

```diff
@@ -0,0 +1,119 @@
+        """
+        response = requests.post(
+            url=url,
+            data=json_data,
+            headers=headers
+        )
+        return response
+
+    def _get_enrollment_headers(self):
+        """
+        """
+        headers = {
+            "Authorization": "Bearer {token}".format(token=self.DROPBOX_TOKEN),
+            "Content-Type": "application/octet-stream",
+            "Dropbox-API-Arg": "{\"path\":\"%s\",\"mode\":{\".tag\":\"overwrite\"}}" % self.DROPBOX_FILE_PATH
+        }
+
+        return headers
+
+    def _get_download_headers(self):
+        """
+        """
+        headers = {
+            "Authorization": "Bearer {token}".format(token=self.DROPBOX_TOKEN),
+            "Dropbox-API-Arg": "{\"path\":\"%s\"}" % self.DROPBOX_FILE_PATH
+        }
+
+        return headers
+
+    def _get_enrollment_data(self, data, course_settings):
+        """
+        """
+        user, _ = get_user(email=data.get("user_email"))
+        dropbox_file = self._get_course_list(course_settings)
+        dropbox_file = dropbox_file.text.split('\n')
+        enrollment_data = u"{first_name}, {last_name}, {email}, {course_id}\n".format(
+            first_name=user.first_name,
+            last_name=user.last_name,
+            email=user.email,
+            course_id=course_settings.get("external_course_run_id"),
+        )
+        enrollment_search_pattern = u"{email}, {course_id}".format(
+            email=user.email,
+            course_id=course_settings.get("external_course_run_id"),
+        )
+        enrolled = False
+
+        for enrollment in dropbox_file:
+            if enrollment_search_pattern in enrollment:
+                enrolled = True
+                dropbox_file.remove(enrollment)
+
+        dropbox_file = "\n".join(dropbox_file)
+
+        if not enrolled:
+            dropbox_file += enrollment_data
+
+        temp_file = io.StringIO()
+        temp_file.write(dropbox_file)
+        temp_file.seek(0)
+        return temp_file.getvalue()
+
+    def _get_enrollment_url(self, course_settings):
+        """
+        """
+        return '{root_url}{path}'.format(root_url=self.DROPBOX_API_URL, path='/files/upload')
+
+    def _get_course_list(self, course_settings):
+        url = '{root_url}{path}'.format(root_url=self.DROPBOX_API_URL, path='/files/download')
+        log_details = {
+            "url": url,
+            "course_advanced_settings": course_settings,
+        }
+        try:
+            response = requests.post(url, headers=self._get_download_headers())
+        except Exception as error:
+            LOG.error("Failed to download course list. Reason: %s", str(error))
+            log_details["response"] = {"error": "Failed to download dropbox course list. Reason: " + str(error)}
+            EnrollmentRequestLog.objects.create(
+                request_type=str(self),
+                details=log_details,
+            )
+            return str(error), status.HTTP_400_BAD_REQUEST
```

Comment:
How do you know it is a bad request? I could be a 500 error. @luismorenolopera Please return this value from the response.

Reply:
done